src/preProcessPattern.js

```diff
@@ -13,7 +12,7 @@ export default function preProcessPattern(globalRef, pattern) {
 
     pattern = typeof pattern === 'string' ? {
         from: pattern
-    } : _.cloneDeep(pattern);
+    } : Object.assign({}, pattern);
```

Comment:
This might be ok here, but `Object.assign` doesn't deep clone `{Object}`'s to my knowledge